### ISSUE TITLE
How to connect or use my dataset to put in a CNN LSTM model

### ISSUE THREAD

**Issue by akshat-suwalka** [tensorflow/tensorflow#42359]
I am a beginner and need guidance in writing the code to connect my dataset to the model i found somewhere in Github.

My dataset looks like:

This is How it look in my directory which shows the number of samples i have
![1](https://user-images.githubusercontent.com/35618437/90235614-3f07c080-de3f-11ea-9744-809c3c82d5ef.jpg)

Inside each sample there are 2 category AM/PM in which my model have to finally classify
![2](https://user-images.githubusercontent.com/35618437/90235644-48912880-de3f-11ea-95c6-d4d480405153.jpg)

This is the time series images, I have 8 images with certail time interval
![3](https://user-images.githubusercontent.com/35618437/90235664-5181fa00-de3f-11ea-8251-3423671c9038.jpg)

`epochs = 52`
 `time = 8`
 `n_classes = 2`
 `width,height,color_channels = 210,140,3`
 `number_of_hiddenunits = 32`
` batch_size = 16`

` def get_conv_vgg(self,input_batch):`

    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.Conv2D(64, (3,3), padding='same', activation='relu') )(input_batch)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.BatchNormalization())(conv_model)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.MaxPool2D(pool_size=(2,2), padding='SAME',strides=(2,2)))(conv_model)

    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.Conv2D(128, (3,3), padding='same', activation='relu') )(conv_model)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.BatchNormalization())(conv_model)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.MaxPool2D(pool_size=(2,2), padding='SAME',strides=(2,2)))(conv_model)

    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.Conv2D(256, (3,3), padding='same', activation='relu') )(conv_model)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.Conv2D(256, (3,3), padding='same', activation='relu') )(conv_model)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.BatchNormalization())(conv_model)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.MaxPool2D(pool_size=(2,2), padding='SAME',strides=(2,2)))(conv_model)

    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.Conv2D(512, (3,3), padding='same', activation='relu') )(conv_model)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.Conv2D(512, (3,3), padding='same', activation='relu') )(conv_model)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.Conv2D(512, (3,3), padding='same', activation='relu') )(conv_model)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.BatchNormalization())(conv_model)
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.MaxPool2D(pool_size=(2,2), padding='SAME',strides=(2,2)))(conv_model)

    #embedded
    conv_model = tf.keras.layers.TimeDistributed(tf.keras.layers.Flatten())(conv_model)

    return conv_model`

`def create_network(self,model_name):`
    
    input_batch = tf.keras.layers.Input(shape = (time,height,width,color_channels))
    if model_name == 'vgg':
        image_features = self.get_conv_vgg(input_batch)
        lstm_network = tf.keras.layers.LSTM(number_of_hiddenunits, return_sequences=True,dropout=0.5,recurrent_dropout=0.5)(image_features)
        lstm_network = tf.keras.layers.LSTM(number_of_hiddenunits, return_sequences=False,dropout=0.5,recurrent_dropout=0.5)(lstm_network)
        lstm_network = tf.keras.layers.Dense(1024,activation='relu')(lstm_network)
        lstm_network = tf.keras.layers.BatchNormalization()(lstm_network)
        lstm_network = tf.keras.layers.Dropout(0.5)(lstm_network)
        lstm_network = tf.keras.layers.Dense(512,activation='relu')(lstm_network)
        lstm_network = tf.keras.layers.Dropout(0.5)(lstm_network)
        lstm_network = tf.keras.layers.Dense(64,activation='relu')(lstm_network)
        lstm_network = tf.keras.layers.Dropout(0.5)(lstm_network)    
        lstm_network = tf.keras.layers.Dense(n_classes,activation='softmax')(lstm_network)

    elif model_name == 'inception':
        image_features = self.get_conv_inception(input_batch)
        lstm_network = tf.keras.layers.LSTM(number_of_hiddenunits, return_sequences=True,dropout=0.5,recurrent_dropout=0.5)(image_features)
        lstm_network = tf.keras.layers.LSTM(number_of_hiddenunits, return_sequences=False,dropout=0.5,recurrent_dropout=0.5)(lstm_network)
        lstm_network = tf.keras.layers.Dense(512,activation='relu')(lstm_network)
        lstm_network = tf.keras.layers.Dense(64,activation='relu')(lstm_network)
        lstm_network = tf.keras.layers.Dropout(0.5)(lstm_network)    
        lstm_network = tf.keras.layers.Dense(n_classes,activation='softmax')(lstm_network)

    full_network = tf.keras.Model([input_batch],lstm_network)
    full_network.summary()
    return full_network`

`def _trainer(network,train_generator,val_generator):`

    network.compile(optimizer = 'adam', loss= 'binary_crossentropy',metrics = ['accuracy'])
    network.save_weights(checkpoint_path.format(epoch=0))
    history =network.fit_generator(train_generator,epochs=epochs,steps_per_epoch=len(os.listdir(train_folder)) // batch_size,validation_data=val_generator,validation_steps=1,callbacks=[cp_callback,tensorboard_callback])`

**After seeing above situation my question is how to create train_generator and val_generator if looking my above dataset format

It is CNN LSTM architecture in which the 8 images sequence is passed through vgg and LSTM

My doubt is how can i pass this 8 images of each category in my model to train it**

**Comment by Saduf2019** [comment]
@akshat-suwalka 
This question is better asked on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.
You may refer to these links:
[link](https://www.kaggle.com/tags/lstm) [link1](https://analyticsindiamag.com/how-to-code-your-first-lstm-network-in-keras/)
